### PR TITLE
{Network} New feature for `az network vnet-gateway` and `az network vpn-connection`

### PR DESCRIPTION
**Pull request by msyyc**
**Description**

https://github.com/Azure/azure-cli/issues/17719
**Testing Guide**

NetworkVpnConnectionNatRule
NetworkVirtualNetworkGatewayNatRule
**History Notes**

[Network]`az network vnet-gateway create`: Add new paramter `--nat-rule`
[Network]Add new cmd group `az network vnet-gateway nat-rule`
[Network]`az network vpn-conncetion create`: Add new paramter `--ingress-nat-rule` and `--egress-nat-rule`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 1]
```diff
@@ -5477,6 +5477,20 @@
 helps['network vnet-gateway create'] = """
 type: command
 short-summary: Create a virtual network gateway.
+parameters:
+  - name: --nat-rule
+    short-summary: VirtualNetworkGatewayNatRule Resource.
+    long-summary: |
+        Usage: --nat-rule name=rule type=Static mode=EgressSnat internal-mappings=10.0.0.0,10.0.0.0 external-mappings=10.0.0,10.0.0 ip-config-id=...
```
Comment (msyyc) [inline src/azure-cli/azure/cli/command_modules/network/_help.py:5484]:
```suggestion
        Usage: --nat-rule name=rule type=Static mode=EgressSnat internal-mappings=10.4.0.0/24 external-mappings=192.168.21.0/24
```

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 2]
```diff
@@ -5477,6 +5477,20 @@
 helps['network vnet-gateway create'] = """
 type: command
 short-summary: Create a virtual network gateway.
+parameters:
+  - name: --nat-rule
+    short-summary: VirtualNetworkGatewayNatRule Resource.
+    long-summary: |
+        Usage: --nat-rule name=rule type=Static mode=EgressSnat internal-mappings=10.4.0.0/24 external-mappings=192.168.21.0/24
```
Comment (msyyc) [inline src/azure-cli/azure/cli/command_modules/network/_help.py:5484]:
```suggestion
        Usage: --nat-rule name=rule type=Static mode=EgressSnat internal-mappings=10.4.0.0/24 external-mappings=192.168.21.0/24 ip-config-id=/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworkGateways/gateway1/ipConfigurations/default
```

**Comment by yonzhan** [comment]
Network

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 3]
```diff
@@ -2004,6 +2015,8 @@ def load_arguments(self, _):
         c.argument('routing_weight', type=int, help='Connection routing weight')
         c.argument('use_policy_based_traffic_selectors', min_api='2017-03-01', help='Enable policy-based traffic selectors.', arg_type=get_three_state_flag())
         c.argument('express_route_gateway_bypass', min_api='2018-07-01', arg_type=get_three_state_flag(), help='Bypass ExpressRoute gateway for data forwarding.')
+        c.argument('ingress_nat_rule', nargs='+', help='List of ingress NatRules.', min_api='2021-02-01', is_preview=True)
+        c.argument('egress_nat_rule', nargs='+', help='List of egress NatRules.', min_api='2021-02-01', is_preview=True)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:2019]: Do we need to pass id of nat rule for these arguments? Can you add any examples for them?

File: src/azure-cli/azure/cli/command_modules/network/_template_builder.py [hunk 4]
```diff
@@ -444,7 +444,7 @@ def build_vnet_resource(_, name, location, tags, vnet_prefix=None, subnet=None,
 
 def build_vpn_connection_resource(cmd, name, location, tags, gateway1, gateway2, vpn_type, authorization_key,
                                   enable_bgp, routing_weight, shared_key, use_policy_based_traffic_selectors,
-                                  express_route_gateway_bypass):
+                                  express_route_gateway_bypass, ingress_nat_rule, egress_nat_rule):
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_template_builder.py:447]: Are they required?

File: src/azure-cli/azure/cli/command_modules/network/_params.py (hunk 3 above)
Comment (msyyc) [reply src/azure-cli/azure/cli/command_modules/network/_params.py:2019]: Have added example in '--help' message

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**